### PR TITLE
Inconsistent behavior with multiple annotations in imagery

### LINKED ISSUE #7170: Inconsistent behavior with multiple annotations in imagery
#### Summary
Lots of unexpected whackiness when adding and removing multiple annotations
Some observed behaviors
* when you add multiple annotations, each subsequent annotation adds its own opacity filter so the selected annotation continues to get lighter and lighter
* when you have multiple annotations and delete 1 or more, the selected annotation appears to become unselected, even though it technically still is and you can continue to add or remove annotations from it (noticed for right most annotation but not neccessarily for other annotations)
* different combinations of adding/removing/clicking on annotation can make the tags temporarily dissapear except for latest addition, which come back after selecting the annotation again

#### Steps to Reproduce
Create and navigate to an example imagery
Hold Alt+shift and drag over imagery to create an annotation
When selecting an existing annotation, use the inspector to add tags

Multiple annotations changes selected annotation color
1. add an annotation
2. add a tag and observe the annotation becomes selected (turns purple if not already from yellow)
3. add more tags and observe the selected annotation becomes lighter on

Annotation appears deselected
1. add an annotation
2. add two tags
3. delete the right-most tag and observe the annotation appears to become deselected (turns yellow from purple)

Tags appear to disappear but are still there
1. add an annotation
2. add a tag
3. re-select the annotation
4. add another tag and observe original tag dissapears
5. re-select the annotation and observe all the annotations reappear
6. rinse and repeat with variations to see all kinds of fun behavior

#### Environment

* Open MCT Version: 
* Deployment Type: 
* OS:
* Browser:

#### Impact Check List

- [x] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [x] Is there a workaround available?
- [ ] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by scottbell**
Closes  #7170 

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/plugins/imagery/components/AnnotationsCanvas.vue [hunk 1]
```diff
@@ -296,9 +316,20 @@ export default {
     cancelSelection(event) {
       if (this.$refs.canvas) {
         const clickedInsideCanvas = this.$refs.canvas.contains(event.target);
+        const tagEditorClassNames = [
```
Comment (scottbell) [inline src/plugins/imagery/components/AnnotationsCanvas.vue:319]: i really don't like this!

File: src/plugins/imagery/components/AnnotationsCanvas.vue [hunk 2]
```diff
@@ -33,6 +33,7 @@
 
 <script>
 import Flatbush from 'flatbush';
+import _ from 'lodash';
```
Comment (davetsay) [inline src/plugins/imagery/components/AnnotationsCanvas.vue:36]:
```suggestion
import isEqual from 'lodash/isequal';
```

File: src/plugins/imagery/components/AnnotationsCanvas.vue [hunk 3]
```diff
@@ -164,7 +178,13 @@ export default {
       const targetDetails = [];
       annotations.forEach((annotation) => {
         annotation.targets.forEach((target) => {
-          targetDetails.push(toRaw(target));
+          // only add targetDetails if we haven't added it before
+          const targetAlreadyAdded = targetDetails.some((targetDetail) => {
+            return _.isEqual(targetDetail, toRaw(target));
```
Comment (davetsay) [inline src/plugins/imagery/components/AnnotationsCanvas.vue:183]:
```suggestion
            return isEqual(targetDetail, toRaw(target));
```
see above for explicitly importing `lodash/isequal`

**Review by davetsay — CHANGES_REQUESTED** [review]
Couple minor code cleanup suggestions. Good work

File: src/plugins/inspectorViews/annotations/tags/TagEditorClassNames.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const TagEditorClassNames = Object.freeze({
```
Comment (davetsay) [inline src/plugins/inspectorViews/annotations/tags/TagEditorClassNames.js:1]: 👍

File: e2e/tests/functional/plugins/imagery/exampleImagery.e2e.spec.js [hunk 5]
```diff
@@ -247,6 +247,14 @@ test.describe('Example Imagery Object', () => {
...
     await expect(page.getByText('Driving')).toBeVisible();
     await expect(page.getByText('Science')).toBeVisible();
+
+    // add another tag and expect it to appear without changing selection
+    await page.getByRole('button', { name: /Add Tag/ }).click();
+    await page.getByPlaceholder('Type to select tag').click();
+    await page.getByText('Drilling').click();
+    await expect(page.getByText('Driving')).toBeVisible();
+    await expect(page.getByText('Science')).toBeVisible();
+    await expect(page.getByText('Drilling')).toBeVisible();
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/imagery/exampleImagery.e2e.spec.js:257]: nice

**Review by unlikelyzero — APPROVED** [review]

**Review by davetsay — APPROVED** [review]
Good stuff

**Pull request merged**